tournamentcontrol/competition/tests/test_thumbnails.py

```diff
@@ -0,0 +1,176 @@
+        self.assertEqual(upload.size(), len(self.image_data))
+        self.assertFalse(upload.resumable())
+        self.assertTrue(upload.has_stream())
+    
+    @patch('magic.from_buffer')
+    def test_media_memory_upload_resumable(self, mock_magic):
+        """Test creating resumable MediaMemoryUpload."""
+        mock_magic.return_value = self.mimetype
+        upload = MediaMemoryUpload(self.image_data, resumable=True)
+        self.assertTrue(upload.resumable())
+    
+    @patch('magic.from_buffer')
+    def test_getbytes(self, mock_magic):
+        """Test getting bytes from upload."""
+        mock_magic.return_value = self.mimetype
+        upload = MediaMemoryUpload(self.image_data)
+        
+        # Get first 10 bytes
+        chunk = upload.getbytes(0, 10)
+        self.assertEqual(chunk, self.image_data[:10])
+        
+        # Get middle chunk
+        chunk = upload.getbytes(5, 15)
+        self.assertEqual(chunk, self.image_data[5:15])
+    
+    @patch('magic.from_buffer')
+    def test_auto_detect_mimetype(self, mock_magic):
+        """Test auto-detecting MIME type when creating MediaMemoryUpload."""
+        mock_magic.return_value = 'image/png'
+        upload = MediaMemoryUpload(self.image_data)
+        
+        self.assertIsNotNone(upload)
+        self.assertEqual(upload.mimetype(), 'image/png')
+        mock_magic.assert_called_once_with(self.image_data, mime=True)
+
+
+class SeasonThumbnailTestCase(TestCase):
+    """Test Season thumbnail functionality."""
+    
+    def setUp(self):
+        self.season = SeasonFactory()
+        self.image_data = b'fake image data'
+        self.mimetype = 'image/jpeg'
+    
+    def test_set_thumbnail_image(self):
+        """Test setting thumbnail image on season."""
+        self.season.set_thumbnail_image(self.image_data)
+        
+        self.season.refresh_from_db()
+        self.assertEqual(self.season.live_stream_thumbnail_image, self.image_data)
+    
+    @patch('magic.from_buffer')
+    def test_get_thumbnail_media_upload(self, mock_magic):
+        """Test getting MediaMemoryUpload from season."""
+        mock_magic.return_value = self.mimetype
+        
+        self.season.live_stream_thumbnail_image = self.image_data
+        self.season.save()
+        
+        upload = self.season.get_thumbnail_media_upload()
+        
+        self.assertIsNotNone(upload)
+        self.assertIsInstance(upload, MediaMemoryUpload)
+        self.assertEqual(upload.mimetype(), self.mimetype)
+    
+    def test_get_thumbnail_media_upload_no_image(self):
+        """Test getting MediaMemoryUpload when no image is set."""
+        upload = self.season.get_thumbnail_media_upload()
+        
+        self.assertIsNone(upload)
+
+
+class MatchThumbnailTestCase(TestCase):
+    """Test Match thumbnail functionality."""
+    
+    def setUp(self):
+        self.season = SeasonFactory()
+        self.match = MatchFactory(stage__division__season=self.season)
```

Comment:
Need to call `.create()` on the factories.

The Season would be implicitly created, this could be a single statement.